pkg/sql/catalog/lease/descriptor_state.go

```diff
@@ -172,6 +172,7 @@ func (t *descriptorState) upsertLeaseLocked(
 	}
 	if session != nil {
 		s.mu.lease.sessionID = session.ID().UnsafeBytes()
+		toRelease = nil
```

Comment:
Add a comment stating why toRelease is being cleared here.

// When using session based leasing, if we end up acquiring the same lease again nothing needs to be cleaned up or updated. This is because the system.lease table does not store any expiry inside the table.

Reply:
Thanks Faizan! I added the comment.
<!-- Reviewable comment -OEVf0r_CQ8wJZW4LnXA:bal6i1r -->
<!-- Sent from Reviewable.io -->